### DISCUSSION TITLE
ImageMagick not detected despite being on Kubuntu system

### DISCUSSION THREAD

**Discussion by MMachado05** [ReagentX/imessage-exporter#328] [Q&A] [answered]
I have ImageMagick installed as a deb package on my Kubuntu system, but when trying to make an export with compatibility turned on, it says it couldn't find an HEIC converter. I also tried installing the AppImage and adding it to my `$PATH`, but still, no dice. Is there some way I can confirm that imessage-exporter is searching for ImageMagick in the right place?

**Comment by ReagentX** [comment]
On non-Windows platforms, `imessage-exporter` checks the output of the [`type`](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#index-type) command to determine if it can use ImageMagick:

https://github.com/ReagentX/imessage-exporter/blob/511d17d217ab4d7708632ec0a706db09eeb4f12d/imessage-exporter/src/app/converter.rs#L47-L62

When you run `imessage-exporter` do you get the `No HEIC converter found` log? Or does it fail to convert images altogether?

**Reply by MMachado05** [reply]
I get the `No HEIC converter found` log, and it says that it won't convert the images.

Weirdly, however, despite APT telling me I have the latest ImageMagick version installed, trying to run the `magick` command doesn't do anything. It may be that something is strangely configured in my system.

**Reply by ReagentX** [reply]
If you figure out why, can you post what you found here? It may help other people in the future.

**Reply by MMachado05** [reply]
I will! Uninstalling and reinstalling didn't do anything... :( I'll keep trying, though!

**Reply by MMachado05** [reply, marked as answer]
I *may* have found something; on *Ubuntu 22.04, the version of ImageMagick installed is from the 6.x series, which doesn't yet unify all the commands under the `magick` umbrella. Does imessage-exporter expect a version 7.x or above, @ReagentX?

**Reply by ReagentX** [reply]
Yes; v7 was released in 2016.

**Reply by MMachado05** [reply]
Why in the world is Ubuntu shipping with such an obscenely out-of-date binary... 💀

Well, at least we know what the issue is... I suppose an upgrade might be in order!
